src/v/raft/configuration_manager.cc

```diff
@@ -89,25 +88,48 @@ ss::future<> configuration_manager::prefix_truncate(model::offset offset) {
          *                            └───────┘
          *                   move last know configuration to truncate offset
          *
+         * Another situation is when prefix truncate happen in between the
+         * configurations f.e:
+         *            200
+         *             │
+         *             │
+         *             │ prefix truncate
+         *             │
+         * ┌───────┐   ▼     ┌───────┐
+         * │   0   │........ │  800  │...........................
+         * └───────┘         └───────┘
+         *     │                      ┌───────┐       ┌───────┐
+         *     └─────────────────────►│  200  │.......│  800  │....
+         *                            └───────┘       └───────┘
+         *
          *  NOTE: box with number represent an entry in configuration manager
          */
 
-        if (it == _configurations.end()) {
-            auto config = std::move(std::prev(it)->second);
-            _configurations.clear();
-            _configurations.emplace(offset, std::move(config));
-        } else {
-            _configurations.erase(_configurations.begin(), it);
+        // special case, do nothig if we are asked to truncate before or exactly
+        // at the beggining
+        if (_configurations.begin()->first >= offset) {
+            return ss::now();
         }
 
+        auto it = _configurations.upper_bound(offset);
+
+        auto config = std::move(std::prev(it)->second);
+        _configurations.erase(_configurations.begin(), it);
```

Comment:
if std::prev(it) is equal to offset then this erases it and reinserts the same entry, right?

Reply:
exactly